impl/src/main/kotlin/nl/nuts/consent/bridge/pipelines/CordaStateMachineToNatsPipeline.kt

```diff
@@ -0,0 +1,152 @@
+ * Stopping subscriptions is to make sure retries are not wasted on during a connection problem (just 1)
+ */
+@Service
+class CordaStateMachineToNatsPipeline {
+    private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+
+    @Autowired
+    lateinit var natsManagedConnectionFactory: NatsManagedConnectionFactory
+    lateinit var natsManagedConnection: NatsManagedConnection
+
+    @Autowired
+    lateinit var cordaManagedConnectionFactory: CordaManagedConnectionFactory
+    lateinit var cordaManagedConnection: CordaManagedConnection
+
+    lateinit var masterSlaveConnection: MasterSlaveConnection
+    lateinit var cordaService: CordaService
+
+    @Autowired
+    lateinit var eventStateStore: EventStateStore
+
+    var subscription: Subscription? = null
+
+    /**
+     * Setup the masterSlave connection manager
+     */
+    @PostConstruct
+    fun init() {
+        natsManagedConnection = natsManagedConnectionFactory.`object`
+        cordaManagedConnection = cordaManagedConnectionFactory.`object`
+
+        masterSlaveConnection = MasterSlaveConnection(natsManagedConnection, cordaManagedConnection)
+
+        cordaManagedConnection.name = "smObserver"
+        natsManagedConnection.name = "smPublisher"
+        cordaManagedConnection.onConnected = { startListeners() }
+        cordaManagedConnection.onDisconnected = { stopListeners() }
+
+        masterSlaveConnection.connect()
+    }
+
+    private fun startListeners() {
+        try {
+            val feed = cordaManagedConnection.getConnection().proxy.stateMachinesFeed()
+
+            val observable = feed.updates
+
+            // transfer feed events to callback
+            subscription = observable.subscribe({ update ->
+                val event = eventStateStore.get(update.id.uuid)
+
+                if (event == null) {
+                    logger.debug("Ignoring ${update.id} state machine update")
+                } else {
+                    logger.debug("Processing ${update.id} state machine update")
+                    if (update is StateMachineUpdate.Removed) {
+                        update.result.doOnFailure(Consumer {
+                            val os = ByteArrayOutputStream()
+                            val pw = PrintWriter(os)
+                            pw.println(it.message)
+                            it.printStackTrace(pw)
+                            pw.close()
+                            os.close()
+                            event.error = os.toString()
+
+                            // publish with new state
+                            val jsonBytes = Serialization.objectMapper().writeValueAsBytes(event)
+                            publish("consentRequestErrored", jsonBytes)
+                        })
+                        update.result.doOnSuccess(Consumer {
+                            // update event to suc6 status
+                            // event.name = EventName.EventConsentRequestFlowSuccess
```

Comment:
todo?

Reply:
That event was removed from the state diagram, it was a dead-end state. Is always overridien by another state.